server/src/test/java/org/opensearch/common/settings/SettingTests.java

```diff
@@ -201,7 +201,7 @@ public void testMemorySize() {
         assertEquals(new ByteSizeValue(12), value.get());
 
         assertTrue(settingUpdater.apply(Settings.builder().put("a.byte.size", "20%").build(), Settings.EMPTY));
-        assertEquals(new ByteSizeValue((int) (JvmInfo.jvmInfo().getMem().getHeapMax().getBytes() * 0.2)), value.get());
+        assertEquals(new ByteSizeValue((long) (JvmInfo.jvmInfo().getMem().getHeapMax().getBytes() * 0.2)), value.get());
```

Comment:
Btw, what is the type of `JvmInfo.jvmInfo().getMem().getHeapMax().getBytes() * 0.2`? Does it need a cast at all?

Reply:
JvmInfo.jvmInfo().getMem().getHeapMax().getBytes() * 0.2 is double type, it can't be automatically cast from double to long, unless force cast.